src/entity/BulletY.java

```diff
@@ -13,7 +13,7 @@
  *
  */
 public class BulletY extends Entity {
-
+    Color[] colors = {Color.WHITE, Color.YELLOW, Color.GREEN, Color.MAGENTA, Color.ORANGE, Color.PINK, Color.LIGHT_GRAY};
```

Comment:
[P3]

I think it would be better to move this part into the setSprite method.
Thank you for your efforts!